src/targets/gpu/jit/concat_past_present.cpp

```diff
@@ -88,15 +87,14 @@ struct concat_past_present_compiler : compiler<concat_past_present_compiler>
         auto gqa_params_str = params.make_init_str();
 
         hip_compile_options options;
-        options.set_launch_params(v,
-                                  compute_global_for(ctx,
-                                                     2 * params.batch_size * params.kv_num_heads *
-                                                         params.sequence_length *
-                                                         params.head_size));
+        options.set_launch_params(
+            v,
+            compute_global_for(ctx,
+                               params.batch_size * params.kv_num_heads * params.sequence_length *
+                                   params.head_size));
         options.inputs      = inputs;
-        options.output      = inputs.front();
+        options.output      = inputs.back();
```

Comment:
this deliberate front() to back() change for output?

Reply:
Yes. This is to support the new behavior of only modifying one of the kv-cache tensors and then outputting it.